### DISCUSSION TITLE
ESP32-C3 and WS with mongoose

### DISCUSSION THREAD

**Discussion by Salem-Tho** [cesanta/mongoose#1877] [Ask questions, get answers] [answered]
Hello Community!

I use en ESP32 and the library mongoose to develop my communication web socket, and it works well! Now I would like to use the new ESP processor ESP32-C3! This MCU is a single core with RISK V architecture (very different from the old ESP32)
After fixing some configuration problem and build error, I flash mu code in the ESP32-C3 but it does not work! and I have on my code new behaviour!
My question : The library Mongoose and WS application are compatible with ESP32-C3 architecture?

**Comment by scaprile** [comment, marked as answer]
The Websocket examples depend on Mongoose, not on whatever Mongoose runs on.
For ESP32, Mongoose interfaces with the IDF, and we currently have tested up to a reasonably recent version of it; try one of the ESP32 examples available and let us know. https://github.com/cesanta/mongoose/tree/master/examples/esp32
Please be more specific on the use of terms like "problem", "error", and "does not work"; if something is not working in the examples please open an issue and follow the template.
